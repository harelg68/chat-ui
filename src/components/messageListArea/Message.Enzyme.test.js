/**
 * Created by harelg on 01/02/2018.
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Message from './Message';

function setup(deltaProps) {
  const props = {
    message: {
      avatar: "dog.png",
      username: "Test user",
      text: "Test message...",
      timestamp: 1517510731694
    },
    isOwner: true
  };
  return shallow(<Message {...(Object.assign({},props,deltaProps))} />);
}

// Testing <Message> component rendering.
describe('Testing <Message> component rendering', () => {
  it('should be rendered correctly', () => {
    const wrapper = setup(null);

    expect(wrapper.find('.clsMsgItem').props().className).toEqual(expect.stringContaining('clsOwnerMe'));
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have the correct style for own messages', () => {
    const wrapper = setup({isOwner: true});

    const userNameClass = wrapper.find('.clsMsgItem').props().className;
    expect(userNameClass).toEqual(expect.stringContaining('clsOwnerMe'));
    expect(userNameClass).not.toEqual(expect.stringContaining('clsOwnerOther'));
  });

  it('should have the correct style for others messages', () => {
    const wrapper = setup({isOwner: false});

    const userNameClass = wrapper.find('.clsMsgItem').props().className;
    expect(userNameClass).not.toEqual(expect.stringContaining('clsOwnerMe'));
    expect(userNameClass).toEqual(expect.stringContaining('clsOwnerOther'));
  });
});
