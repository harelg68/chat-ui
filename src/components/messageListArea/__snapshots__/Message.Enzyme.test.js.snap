// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
<ListItem>
  <ListContent>
    <div
      className="clsMsgNormal"
    >
      <div
        className="clsMsgTime"
      >
        02/01/18 20:45:31
      </div>
      <img
        className="clsMsgIcon"
        src="https://spotim-demo-chat-server.herokuapp.com/avatars/004-jigglypuff.png"
      />
      <div
        className="clsMsgItem clsOwnerMe"
      >
        User #3
      </div>
      <div
        className="clsMsgText"
      >
        test message................................. (#2)
      </div>
    </div>
  </ListContent>
</ListItem>
`;

exports[`Testing <Message> component rendering should be rendered correctly 1`] = `
<ListItem>
  <ListContent>
    <div
      className="clsMsgNormal"
    >
      <div
        className="clsMsgTime"
      >
        02/01/18 20:45:31
      </div>
      <img
        className="clsMsgIcon"
        src="dog.png"
      />
      <div
        className="clsMsgItem clsOwnerMe"
      >
        Test user
      </div>
      <div
        className="clsMsgText"
      >
        Test message...
      </div>
    </div>
  </ListContent>
</ListItem>
`;

exports[`Testing <Message> component renders correctly 1`] = `
<ListItem>
  <ListContent>
    <div
      className="clsMsgNormal"
    >
      <div
        className="clsMsgTime"
      >
        02/01/18 20:45:31
      </div>
      <img
        className="clsMsgIcon"
        src="dog.png"
      />
      <div
        className="clsMsgItem clsOwnerMe"
      >
        Test user
      </div>
      <div
        className="clsMsgText"
      >
        Test message...
      </div>
    </div>
  </ListContent>
</ListItem>
`;
